### ISSUE TITLE
7.0.0-alpha2 - JSON serialization error while deserialize composite aggregation

### ISSUE THREAD

**Issue by Phoenix-616** [elastic/elasticsearch-net#3745]
**NEST/Elasticsearch.Net version**: 7.0.0-alpha2

**Elasticsearch version**: 7.0.0

**Description of the problem**:
Getting the following error when trying to do a composite aggregation with single term subaggregation: `Elasticsearch.Net.JsonParsingException: expected:'String Begin Token', actual:',', at offset:340`

**Intercepted request**:
```
{
        "size": 0,
	"aggs": {
		"FieldNamecomposite": {
			"composite": {
				"size": 10,
				"sources": [{
						"FieldNamesub": {
							"terms": {
								"field": "FieldName"
							}
						}
					}
				]
			}
		}
	}
}
```

**Intercepted response**:
```
{
	"took": 28,
	"timed_out": false,
	"_shards": {
		"total": 5,
		"successful": 5,
		"skipped": 0,
		"failed": 0
	},
	"hits": {
		"total": {
			"value": 97,
			"relation": "eq"
		},
		"max_score": null,
		"hits": []
	},
	"aggregations": {
		"composite#FieldNamecomposite": {
			"after_key": {
				"FieldNamesub": 1088442000000
			},
			"buckets": [{
					"key": {
						"FieldNamesub": 1081270800000
					},
					"doc_count": 2
				}, {
					"key": {
						"FieldNamesub": 1081702800000
					},
					"doc_count": 2
				}, {
					"key": {
						"FieldNamesub": 1082307600000
					},
					"doc_count": 2
				}, {
					"key": {
						"FieldNamesub": 1084294800000
					},
					"doc_count": 3
				}, {
					"key": {
						"FieldNamesub": 1084467600000
					},
					"doc_count": 2
				}, {
					"key": {
						"FieldNamesub": 1084813200000
					},
					"doc_count": 3
				}, {
					"key": {
						"FieldNamesub": 1085504400000
					},
					"doc_count": 1
				}, {
					"key": {
						"FieldNamesub": 1087232400000
					},
					"doc_count": 3
				}, {
					"key": {
						"FieldNamesub": 1087750800000
					},
					"doc_count": 1
				}, {
					"key": {
						"FieldNamesub": 1088442000000
					},
					"doc_count": 4
				}
			]
		}
	}
}
```

**Provide `DebugInformation`**:
```
# Audit exception in step 1 BadResponse:
Elasticsearch.Net.JsonParsingException: expected:'String Begin Token', actual:',', at offset:340
   at Elasticsearch.Net.JsonReader.ReadStringSegmentCore(Byte[]& resultBytes, Int32& resultOffset, Int32& resultLength)
   at Elasticsearch.Net.JsonReader.ReadString()
   at Nest.AggregateFormatter.GetCompositeBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.GetKeyedBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.ReadBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.GetMultiBucketAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver, ArraySegment`1& propertyName)
   at Nest.AggregateFormatter.ReadAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateDictionaryFormatter.ReadAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver, String[] tokens, Dictionary`2 dictionary)
   at Nest.AggregateDictionaryFormatter.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Deserialize(Object[] , JsonReader& , IJsonFormatterResolver )
   at Elasticsearch.Net.JsonSerializer.Deserialize[T](Byte[] bytes, Int32 offset, IJsonFormatterResolver resolver)
   at Elasticsearch.Net.JsonSerializer.DeserializeAsync[T](Stream stream, IJsonFormatterResolver resolver)
   at Elasticsearch.Net.ResponseBuilder.SetBodyAsync[TResponse](ApiCallDetails details, RequestData requestData, Stream responseStream, String mimeType, CancellationToken cancellationToken)
   at Elasticsearch.Net.ResponseBuilder.ToResponseAsync[TResponse](RequestData requestData, Exception ex, Nullable`1 statusCode, IEnumerable`1 warnings, Stream responseStream, String mimeType, CancellationToken cancellationToken)
   at Elasticsearch.Net.HttpConnection.RequestAsync[TResponse](RequestData requestData, CancellationToken cancellationToken)
   at Elasticsearch.Net.RequestPipeline.CallElasticsearchAsync[TResponse](RequestData requestData, CancellationToken cancellationToken)
   at Elasticsearch.Net.Transport`1.RequestAsync[TResponse](HttpMethod method, String path, CancellationToken cancellationToken, PostData data, IRequestParameters requestParameters)
# Inner Exception: expected:'String Begin Token', actual:',', at offset:340
Elasticsearch.Net.JsonParsingException: expected:'String Begin Token', actual:',', at offset:340
   at Elasticsearch.Net.JsonReader.ReadStringSegmentCore(Byte[]& resultBytes, Int32& resultOffset, Int32& resultLength)
   at Elasticsearch.Net.JsonReader.ReadString()
   at Nest.AggregateFormatter.GetCompositeBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.GetKeyedBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.ReadBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.GetMultiBucketAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver, ArraySegment`1& propertyName)
   at Nest.AggregateFormatter.ReadAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateDictionaryFormatter.ReadAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver, String[] tokens, Dictionary`2 dictionary)
   at Nest.AggregateDictionaryFormatter.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Deserialize(Object[] , JsonReader& , IJsonFormatterResolver )
   at Elasticsearch.Net.JsonSerializer.Deserialize[T](Byte[] bytes, Int32 offset, IJsonFormatterResolver resolver)
   at Elasticsearch.Net.JsonSerializer.DeserializeAsync[T](Stream stream, IJsonFormatterResolver resolver)
   at Elasticsearch.Net.ResponseBuilder.SetBodyAsync[TResponse](ApiCallDetails details, RequestData requestData, Stream responseStream, String mimeType, CancellationToken cancellationToken)
   at Elasticsearch.Net.ResponseBuilder.ToResponseAsync[TResponse](RequestData requestData, Exception ex, Nullable`1 statusCode, IEnumerable`1 warnings, Stream responseStream, String mimeType, CancellationToken cancellationToken)
   at Elasticsearch.Net.HttpConnection.RequestAsync[TResponse](RequestData requestData, CancellationToken cancellationToken)
   at Elasticsearch.Net.RequestPipeline.CallElasticsearchAsync[TResponse](RequestData requestData, CancellationToken cancellationToken)
   at Elasticsearch.Net.Transport`1.RequestAsync[TResponse](HttpMethod method, String path, CancellationToken cancellationToken, PostData data, IRequestParameters requestParameters)
# Exception:
Elasticsearch.Net.UnexpectedElasticsearchClientException: expected:'String Begin Token', actual:',', at offset:340 ---> Elasticsearch.Net.JsonParsingException: expected:'String Begin Token', actual:',', at offset:340
   at Elasticsearch.Net.JsonReader.ReadStringSegmentCore(Byte[]& resultBytes, Int32& resultOffset, Int32& resultLength)
   at Elasticsearch.Net.JsonReader.ReadString()
   at Nest.AggregateFormatter.GetCompositeBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.GetKeyedBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.ReadBucket(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateFormatter.GetMultiBucketAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver, ArraySegment`1& propertyName)
   at Nest.AggregateFormatter.ReadAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Nest.AggregateDictionaryFormatter.ReadAggregate(JsonReader& reader, IJsonFormatterResolver formatterResolver, String[] tokens, Dictionary`2 dictionary)
   at Nest.AggregateDictionaryFormatter.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Deserialize(Object[] , JsonReader& , IJsonFormatterResolver )
   at Elasticsearch.Net.JsonSerializer.Deserialize[T](Byte[] bytes, Int32 offset, IJsonFormatterResolver resolver)
   at Elasticsearch.Net.JsonSerializer.DeserializeAsync[T](Stream stream, IJsonFormatterResolver resolver)
   at Elasticsearch.Net.ResponseBuilder.SetBodyAsync[TResponse](ApiCallDetails details, RequestData requestData, Stream responseStream, String mimeType, CancellationToken cancellationToken)
   at Elasticsearch.Net.ResponseBuilder.ToResponseAsync[TResponse](RequestData requestData, Exception ex, Nullable`1 statusCode, IEnumerable`1 warnings, Stream responseStream, String mimeType, CancellationToken cancellationToken)
   at Elasticsearch.Net.HttpConnection.RequestAsync[TResponse](RequestData requestData, CancellationToken cancellationToken)
   at Elasticsearch.Net.RequestPipeline.CallElasticsearchAsync[TResponse](RequestData requestData, CancellationToken cancellationToken)
   at Elasticsearch.Net.Transport`1.RequestAsync[TResponse](HttpMethod method, String path, CancellationToken cancellationToken, PostData data, IRequestParameters requestParameters)
   --- End of inner exception stack trace ---
   at Elasticsearch.Net.Transport`1.RequestAsync[TResponse](HttpMethod method, String path, CancellationToken cancellationToken, PostData data, IRequestParameters requestParameters)
```

**Comment by codebrain** [comment]
Fix has been merged and should be available in the next release of the 7.0.0 client.

**Comment by Phoenix-616** [comment]
Great! Thank you!
